File: javascript/1.Two-Sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (nums, target) {
  const map = new Map();

  for (let i = 0; i < nums.length; i++) {
    const req = target - nums[i];

    if (map.has(req)) {
      return [i, map.get(req)];
    }

    map.set(nums[i], i);
  }
};

File: python/1.Two-Sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        result = []
        index_map = {}

        for i in range(len(nums)):
            req = target - nums[i]

            if req in index_map:
                return [i, index_map[req]]

            index_map[nums[i]] = i

        return []